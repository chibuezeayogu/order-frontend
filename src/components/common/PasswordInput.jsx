import React from 'react'

const PasswordInput = ({ value, onChange }) => {
  return (
    <input
      type="password"
      className="form-control rounded-left"
      value={value}
      onChange={onChange}
      id="password"
      placeholder="Password"
    />
  )
}

export default PasswordInput
