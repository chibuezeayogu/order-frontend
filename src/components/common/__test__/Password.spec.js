import React from 'react'
import { shallow } from 'enzyme'
import PasswordInput from '../PasswordInput'

const setUp = () => {
  const props = {
    value: '',
    onChange: jest.fn(),
    text: 'password'
  }

  return shallow(<PasswordInput {...props} />)
}

describe('PasswordInput Component', () => {
  it('should render without crashing', () => {
    const wrapper = setUp()
    const input = wrapper.find('input')

    expect(wrapper).toMatchSnapshot()
    expect(input).toHaveLength(1)
    expect(input.get(0).props.type).toBe('password')
    expect(input.get(0).props.required).toBeTruthy()
    expect(input.get(0).props.placeholder).toBe('Password')
  })
})
