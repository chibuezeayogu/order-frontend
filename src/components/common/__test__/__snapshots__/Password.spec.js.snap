// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput Component should render without crashing 1`] = `
<input
  className="form-control rounded-left"
  id="password"
  onChange={[MockFunction]}
  placeholder="Password"
  required={true}
  type="password"
  value=""
/>
`;
